web/src/App.vue

```diff
@@ -1,26 +1,65 @@
 <template>
-  <v-app class="dandi-app">
+  <v-app
+    v-if="verifiedServerConnection"
+    class="dandi-app"
+  >
     <AppBar />
-    <v-main>
+    <v-main v-if="connectedToServer">
       <UserStatusBanner />
       <router-view />
       <DandiFooter />
     </v-main>
+
+    <v-main
+      v-else
+      class="d-flex align-center text-center"
+    >
+      {{ SERVER_DOWNTIME_MESSAGE }}
+    </v-main>
   </v-app>
 </template>
 
-<script>
+<script lang="ts">
+import { defineComponent, onMounted, ref } from '@vue/composition-api';
+
+import { dandiRest } from '@/rest';
+import store from '@/store';
 import AppBar from '@/components/AppBar/AppBar.vue';
 import DandiFooter from '@/components/DandiFooter.vue';
 import UserStatusBanner from '@/components/UserStatusBanner.vue';
 
-export default {
+const SERVER_DOWNTIME_MESSAGE = process.env.VUE_APP_SERVER_DOWNTIME_MESSAGE || 'Connection to server failed.';
+
+export default defineComponent({
   components: {
     AppBar,
     DandiFooter,
     UserStatusBanner,
   },
-};
+  setup() {
+    const verifiedServerConnection = ref(false);
+    const connectedToServer = ref(true);
+
+    onMounted(() => {
+      Promise.all([
+        store.dispatch.dandiset.fetchSchema(),
+        dandiRest.restoreLogin(),
+      ]).then(() => {
+        connectedToServer.value = true;
+      }).catch(() => {
+        connectedToServer.value = false;
+      }).finally(() => {
+        verifiedServerConnection.value = true;
+      });
+    });
```

Comment:
I don't see the need for two variables here. Couldn't you just use `connectedToServer`, defaulting to `true` and setting to `false` in the `catch` case?

Reply:
I believe that would introduce a race condition. If the server is down or `fetchSchema()`/`restoreLogin()` takes a long time, and just `connectedToServer` is used and defaulted to true, it's possible that `router-view` and it's children will render before `connectedToServer` is set properly. This would result in console errors since nothing in the store is initialized yet.

Reply:
I see. Although in it's current state, it appears you'd just see a blank page when connecting, if your connection was very slow. Even when testing with a deliberately slow connection this was pretty brief though, so I think this is fine.